libpromises/evalfunction.c

```diff
@@ -6180,106 +6180,119 @@ static FnCallResult FnCallReadRealList(EvalContext *ctx, ARG_UNUSED const Policy
     return ReadList(ctx, fp, args, CF_DATA_TYPE_REAL);
 }
 
-static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
-                                   ARG_UNUSED const Policy *policy,
-                                   const FnCall *fp,
-                                   const Rlist *args)
+JsonElement *ReadDataFile(const char* log_identifier, const char* input_path, const char* requested_mode, size_t size_max)
```

Comment:
Wow I just noticed how this one function does everything: JSON, YAML, CSV, ENV. The fine details of parsing should be definitely moved elsewhere. Here the leftovers should only be error checking/logging and calling whatever parsers we have ready.